src/createTokenBridge.ts

```diff
@@ -0,0 +1,298 @@
+import { Address, Chain, PrivateKeyAccount, PublicClient, TransactionReceipt } from 'viem';
+import {
+  CreateTokenBridgeEnoughCustomFeeTokenAllowanceParams,
+  createTokenBridgeEnoughCustomFeeTokenAllowance,
+} from './createTokenBridgeEnoughCustomFeeTokenAllowance';
+import { createTokenBridgePrepareCustomFeeTokenApprovalTransactionRequest } from './createTokenBridgePrepareCustomFeeTokenApprovalTransactionRequest';
+import { TokenBridgeContracts } from './types/TokenBridgeContracts';
+import {
+  TransactionRequestRetryableGasOverrides as TokenBridgeRetryableGasOverrides,
+  createTokenBridgePrepareTransactionRequest,
+} from './createTokenBridgePrepareTransactionRequest';
+import {
+  CreateTokenBridgeTransactionReceipt,
+  WaitForRetryablesResult,
+  createTokenBridgePrepareTransactionReceipt,
+} from './createTokenBridgePrepareTransactionReceipt';
+import {
+  createTokenBridgePrepareSetWethGatewayTransactionRequest,
+  TransactionRequestRetryableGasOverrides as SetWethGatewayRetryableGasOverrides,
+} from './createTokenBridgePrepareSetWethGatewayTransactionRequest';
+import {
+  CreateTokenBridgeSetWethGatewayTransactionReceipt,
+  createTokenBridgePrepareSetWethGatewayTransactionReceipt,
+} from './createTokenBridgePrepareSetWethGatewayTransactionReceipt';
+import { isCustomFeeTokenAddress } from './utils/isCustomFeeTokenAddress';
+import { WithTokenBridgeCreatorAddressOverride } from './types/createTokenBridgeTypes';
+import { TransactionRequestGasOverrides } from './utils/gasOverrides';
+
+function getBlockExplorerUrl(chain: Chain | undefined) {
+  return chain?.blockExplorers?.default.url;
+}
+
+export type CreateTokenBridgeParams = WithTokenBridgeCreatorAddressOverride<{
+  rollupOwner: PrivateKeyAccount;
+  rollupAddress: Address;
+  nativeTokenAddress?: Address;
+  parentChainPublicClient: PublicClient;
+  orbitChainPublicClient: PublicClient;
+  gasOverrides?: TransactionRequestGasOverrides;
+  retryableGasOverrides?: TokenBridgeRetryableGasOverrides;
+  setWethGatewayGasOverrides?: SetWethGatewayRetryableGasOverrides;
+}>;
+export type CreateTokenBridgeReturnType = {
+  /**
+   * Transaction hash of createTokenBridgePrepareTransactionRequest
+   */
+  transaction: Address;
+  /**
+   * Transaction receipt of createTokenBridgePrepareTransactionReceipt ({@link CreateTokenBridgeTransactionReceipt})
+   */
+  transactionReceipt: CreateTokenBridgeTransactionReceipt;
+  /**
+   * Retryable transaction receipts of createTokenBridgePrepareTransactionReceipt ({@link WaitForRetryablesResult})
+   */
+  retryables: WaitForRetryablesResult;
```

Comment:
`WaitForRetryablesResult` resolve to `[TransactionReceipt, TransactionReceipt]`. If we ever update `WaitForRetryablesResult` to something else, then it will be updated here too